src/main/java/edu/cornell/kfs/module/ar/document/web/struts/CuContractsGrantsInvoiceDocumentAction.java

```diff
@@ -41,6 +56,115 @@ public ActionForward prorateBill(ActionMapping mapping, ActionForm form, HttpSer
+        if (contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().isFinalBillIndicator()) {
+            warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_FINAL_BILL_INDICATOR));
+        }
+
+        String billingPeriod = contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getBillingPeriod();
+        if (billingPeriod == null || billingPeriod.length() != 24) {
+            billingPeriod = billingPeriod == null ? KFSConstants.EMPTY_STRING : billingPeriod;
+            warningMessages.add("The Billing Period is Invalid [" + billingPeriod + "]. The expected length is 24 (YYYY-MM-DD to YYYY-MM-DD) do you wish to continue?");
+        } else {
+            try {
+                Pair<Date, Date> billingPeriodDates = parseDateRange(billingPeriod);
+                Date billingPeriodStartDate = billingPeriodDates.getKey();
+                Date billingPeriodEndDate = billingPeriodDates.getValue();
+                DateTimeService dateTimeService = SpringContext.getBean(DateTimeService.class);
+                if (dateTimeService.dateDiff(billingPeriodEndDate, contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getLastBilledDate(), false) >= 1) {
+                    warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_BILLING_PERIOD_END_DATE_BEFORE_LAST_BILLED_DATE));
+                }
+                if (dateTimeService.dateDiff(billingPeriodEndDate, billingPeriodStartDate, false) >= 1) {
+                    warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_BILLING_PERIOD_END_DATE_BEFORE_BILLING_PERIOD_START_DATE));
+                }
+                if (dateTimeService.dateDiff(billingPeriodStartDate, contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getLastBilledDate(), false) >= 1) {
+                    warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_BILLING_PERIOD_START_DATE_BEFORE_LAST_BILLED_DATE));
+                }
+
+                Pair<Date, Date> awardDateRange = parseDateRange(contractsGrantsInvoiceDocument.getInvoiceGeneralDetail().getAwardDateRange());
+                Date awardStartDate = awardDateRange.getKey();
+                if (dateTimeService.dateDiff(billingPeriodStartDate, awardStartDate, false) >= 1) {
+                    warningMessages.add(configurationService.getPropertyValueAsString(CUKFSKeyConstants.WARNING_CINV_BILLING_PERIOD_START_DATE_BEFORE_AWARD_START_DATE));
+                }
+            } catch(ParseException ex) {
+                LOG.error("getContractsGrantsInvoiceDocumentWarningMessage: " + ex.getMessage());
+                warningMessages.add("ParseException occurred while parsing the billing period. Do you want to Proceed?[br]" + ex.getMessage());
+            }
+        }
+
+        String ret = CollectionUtils.isEmpty(warningMessages) ? null : StringUtils.join(warningMessages, "[br]");
```

Comment:
StringUtils.join() should already return a null/empty string for a null/empty list, so you may not need the extra null-check here. Also, it might be good to move the "[br]" separator to its own constant.